### ISSUE TITLE
No checks if disk object already exists on up

### ISSUE THREAD

**Issue by Temikus** [mitchellh/vagrant-google#69]
If we try to create a disk with the same name as already exists, we get a traceback, while it _should_ be processed properly. This needs to be investigated.

Info:
```
λ vagrant up
Your Gemfile lists the gem vagrant-google (>= 0) more than once.
You should probably keep only one of them.
While it's not a problem now, it could cause errors if you change the version of just one of them later.
Vagrant appears to be running in a Bundler environment. Your
existing Gemfile will be used. Vagrant will not auto-load any plugins
installed with `vagrant plugin`. Vagrant will autoload any plugins in
the 'plugins' group in your Gemfile. You can force Vagrant to take over
with VAGRANT_FORCE_BUNDLER.

You appear to be running Vagrant outside of the official installers.
Note that the installers are what ensure that Vagrant has all required
dependencies, and Vagrant assumes that these dependencies exist. By
running outside of the installer environment, Vagrant may not function
properly. To remove this warning, install Vagrant using one of the
official packages from vagrantup.com.

Bringing machine 'default' up with 'google' provider...
==> default: Launching an instance with the following settings...
==> default:  -- Name:            testing-vagrant
==> default:  -- Type:            n1-standard-1
==> default:  -- Disk type:       pd-standard
==> default:  -- Disk size:       10 GB
==> default:  -- Disk name:
==> default:  -- Image:           debian-7-wheezy-v20150127
==> default:  -- Zone:            europe-west1-d
==> default:  -- Network:         default
==> default:  -- Metadata:        '{}'
==> default:  -- Tags:            '[]'
==> default:  -- IP Forward:
==> default:  -- External IP:
==> default:  -- Autodelete Disk: true
==> default:  -- Scopes:    ["768099535741-ucc0c0t9gcemtkv1vea2aknhjmsahksi.apps.googleusercontent.com"]
/home/temikus/.gem/ruby/gems/fog-1.29.0/lib/fog/google/core.rb:202:in `build_excon_response': The resource 'projects/dev-utility-90115/zones/europe-west1-d/disks/testing-vagrant' already exists (Fog::Errors::Error)
	from /home/temikus/.gem/ruby/gems/fog-1.29.0/lib/fog/google/core.rb:173:in `request'
	from /home/temikus/.gem/ruby/gems/fog-1.29.0/lib/fog/google/requests/compute/insert_disk.rb:108:in `insert_disk'
	from /home/temikus/.gem/ruby/gems/fog-1.29.0/lib/fog/google/models/compute/disk.rb:40:in `save'
	from /home/temikus/.gem/ruby/gems/fog-core-1.30.0/lib/fog/core/collection.rb:51:in `create'
	from /home/temikus/code/tcr-vagrant-google/lib/vagrant-google/action/run_instance.rb:98:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/warden.rb:34:in `call'
	from /home/temikus/code/tcr-vagrant-google/lib/vagrant-google/action/warn_networks.rb:28:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/warden.rb:34:in `call'
	from /home/temikus/code/tcr-vagrant-google/lib/vagrant-google/action/sync_folders.rb:33:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/warden.rb:34:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/builtin/provision.rb:80:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/warden.rb:34:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/warden.rb:34:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/warden.rb:34:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/builder.rb:116:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/runner.rb:66:in `block in run'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/util/busy.rb:19:in `busy'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/runner.rb:66:in `run'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/builtin/call.rb:53:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/warden.rb:34:in `call'
	from /home/temikus/code/tcr-vagrant-google/lib/vagrant-google/action/connect_google.rb:47:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/warden.rb:34:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/builtin/box_check_outdated.rb:36:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/warden.rb:34:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/warden.rb:34:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/builtin/handle_box.rb:56:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/warden.rb:34:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/builder.rb:116:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/runner.rb:66:in `block in run'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/util/busy.rb:19:in `busy'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/action/runner.rb:66:in `run'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/machine.rb:214:in `action_raw'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/machine.rb:191:in `block in action'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/environment.rb:516:in `lock'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/machine.rb:178:in `call'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/machine.rb:178:in `action'
	from /home/temikus/.gem/ruby/bundler/gems/vagrant-d7eaf5032980/lib/vagrant/batch_action.rb:82:in `block (2 levels) in run'
```

**Comment by Temikus** [comment]
Fog-google errors are mostly raising generic Fog::Errors::Error exceptions, so if we add a catch-all, this may lead to a similarly bad UI experience, for example:
```
There was an error talking to Google. The error message is shown
below:
CghJTlNUQU5DRRIXMjg4Mjc4OTg1ODI3Lmluc3RhbmNlLTE='
```
Until Fog homogenizes errors (see https://github.com/fog/fog/issues/3190), we'll need to implement disk existence checking.

**Comment by Temikus** [comment]
Fixed by #113
